app.py

```diff
@@ -589,7 +589,18 @@ def endpoint_home():
 #Default formbar in advanced expert mode
 @app.route('/expert')
 def endpoint_expert():
-    return render_template('expert.html')
+    if sD.studentDict:
+        username = sD.studentDict[request.remote_addr]['name']
+    else:
+        username = ''
```

Comment:
Instead of pulling out just the individual properties like "name", you can just send the entire sD.studentDict[request.remote_addr].